integreat_cms/static/src/js/unsaved-warning.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * This file contains a function to warn the user when they leave a content without saving
+ */
+
+ var confirmed = false;
+ var edited = false;
+
+
+ window.addEventListener('beforeunload', function(e) {
+    /*trigger only when something is edited and no submit/save button clicked*/
+   if( edited && !confirmed) {
+     e.preventDefault();
+     e.returnValue = 'This content is not saved. Would you leave the page?';
+   }
+ });
+ 
+
+
+
+window.addEventListener("load", () => {
+    /*checks whether the user typed something in the content*/
+    var form = document.getElementById("content_form");
+    form.addEventListener("input", function(e){
+        
+        edited = true;
+        console.log("editing detected");
+    });
```

Comment:
```suggestion   
    const form = document.getElementById("content_form");
    if (form) {
        form.addEventListener("input", () => {
            edited = true;
            console.log("editing detected");
        });
   }
```

Otherwise there was an Error, when form was null